pkg/appender/lru_cache.go

```diff
@@ -73,7 +64,7 @@ func (c *Cache) removeOldest() {
 		ele := c.free.Back()
 		if ele != nil {
 			c.free.Remove(ele)
-			kv := ele.Value.(*clist.Element).Value.(*entry)
+			kv := ele.Value.(*entry)
```

Comment:
you can save the `key` itself, you don't use `kv`

Reply:
Also, isn't `ele` can be nil after line 66? better get the key before the remove

Reply:
ele.Value won't be nil, remove only deletes the list associated vars (head, prev, next)